File: src/main/java/org/myfintech/payment/service/ContractService.java
package org.myfintech.payment.service;

import java.util.List;
import java.util.Optional;
import java.util.Set;

import org.myfintech.payment.domain.ContractCreateDTO;
import org.myfintech.payment.domain.ContractDTO;
import org.myfintech.payment.entity.Contract;
import org.myfintech.payment.entity.projection.ContractWithClientProjection;

public interface ContractService {
	List<ContractDTO> findAll();

	ContractDTO findById(Long id);

	ContractDTO save(ContractCreateDTO dto);

	ContractDTO update(Long id, ContractDTO dto);

	Optional<Contract> findByContractNumber(String contractNumber);

	List<ContractWithClientProjection> findAllByContractNumbers(Set<String> contractNumbers);

	void validateUpdateRequest(Long id, ContractDTO contractDTO);

	void validateCreate(ContractCreateDTO contractDTO);
}

File: src/main/kotlin/org/myfintech/payment/service/ContractService.kt
package org.myfintech.payment.service

import org.myfintech.payment.domain.ContractCreateDTO
import org.myfintech.payment.domain.ContractDTO
import org.myfintech.payment.entity.Contract
import org.myfintech.payment.entity.projection.ContractWithClientProjection
import java.util.Optional

// ... other imports

interface ContractService {
    fun findAll(): List<ContractDTO>
    fun findById(id: Long): ContractDTO
    fun save(dto: ContractCreateDTO): ContractDTO
    fun update(id: Long, dto: ContractDTO): ContractDTO

    fun findByContractNumber(contractNumber: String): Contract
    fun findAllByContractNumbers(contractNumbers: Set<String>): List<ContractWithClientProjection>
    fun validateUpdateRequest(id: Long, contractDTO: ContractDTO)
    fun validateCreate(contractDTO: ContractCreateDTO)
}